daemon/actual_ip_resolver.go

```diff
@@ -0,0 +1,131 @@
+package daemon
+
+import (
+	"context"
+	"log"
+	"net/netip"
+	"time"
+
+	"github.com/NordSecurity/nordvpn-linux/core"
+	"github.com/NordSecurity/nordvpn-linux/daemon/state"
+	"github.com/NordSecurity/nordvpn-linux/events"
+	"github.com/NordSecurity/nordvpn-linux/internal"
+	"github.com/NordSecurity/nordvpn-linux/network"
+)
+
+// tryInsightsIP is an attempt to get insights IP
+// the IP it returns is always valid
+func tryInsightsIP(api core.InsightsAPI) (netip.Addr, error) {
+	insights, err := api.InsightsViaTunnel()
+	if err == nil && insights != nil {
+		ip, err := netip.ParseAddr(insights.IP)
+		if err == nil {
+			return ip, nil
+		} else {
+			return netip.Addr{}, err
+		}
+	} else {
+		return netip.Addr{}, err
+	}
```

Comment:
This can be simplified - use DeMorgan's law to invert the condition and return early, this way you can get rid of nesting. It creates a precondition for what happens next and it's easier to reason about the code, because at the top you see required conditions and then when conditions are met, you see the meat of the function

So instead of
```
if err == nil && insights != nil {
...
} else {
...
}
```
you can have:
```
if err != nil || insights == nil {
  // return early
}
// no else needed - less nesting
```

It can be done also for internal `if` block